### PR TITLE
(2/17) Creates `XRPAccountDelete` Protobuf Wrapper Class

### PR DESCRIPTION
**Pull request by amiecorso**
## High Level Overview of Change
This PR implements the class `XRPAccountDelete` - a native Typescript class to wrap the [`AccountDelete` protocol buffer](https://github.com/ripple/rippled/blob/3d86b49dae8173344b39deb75e53170a9b6c5284/src/ripple/proto/org/xrpl/rpc/v1/transaction.proto#L118).  See also: https://xrpl.org/accountdelete.html.

### Context of Change
Effort to implement additional [XRPL transaction types](https://xrpl.org/transaction-types.html) beyond `Payment` in the SDK.

### Type of Change

- [x] New feature (non-breaking change which adds functionality)
- [x] Tests (You added tests for code that already exists, or your new feature included in this PR)

## Before / After
`XRPAccountDelete` class and conversion unit tests now exist.

## Test Plan
Unit tests added to verify conversion.  CI passes.

## Future Tasks

Remaining transaction types & all three SDKs.
Implementation and exposure of client methods for submitting additional transaction types.

### CODE REVIEW

File: src/XRP/model/xrp-account-delete.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+   *                to construct an XRPAccountDelete
+   * @return an XRPAccountDelete with its fields set via the analogous protobuf fields.
+   * @see https://github.com/ripple/rippled/blob/3d86b49dae8173344b39deb75e53170a9b6c5284/src/ripple/proto/org/xrpl/rpc/v1/transaction.proto#L118
+   */
+  public static from(
+    accountDelete: AccountDelete,
+  ): XRPAccountDelete | undefined {
+    // TODO(amiecorso): decide whether the existence of `destination` needs to be policed here
+    //                  via returning undefined or throwing an error
+    // TODO(amiecorso): should this object have an X-address version of this information?
```
Comment (amiecorso) [inline src/XRP/model/xrp-account-delete.ts:23]: Note these TODOs... I'd like to get all of these sorts of details settled before moving on to the other SDKs.

File: src/XRP/model/xrp-account-delete.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
+import { AccountDelete } from '../Generated/web/org/xrpl/rpc/v1/transaction_pb'
+
+/*
+ * Represents an AccountDelete transaction on the XRP Ledger.
+ * An AccountDelete transaction deletes an account and any objects it owns in the XRP Ledger,
```
Comment (keefertaylor) [inline src/XRP/model/xrp-account-delete.ts:5]: +1 blank line

File: src/XRP/model/xrp-account-delete.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    //                  via returning undefined or throwing an error
+    // TODO(amiecorso): should this object have an X-address version of this information?
+    const destination = accountDelete.getDestination()?.getValue()?.getAddress()
+    const destinationTag = accountDelete.getDestinationTag()?.getValue()
+
+    return new XRPAccountDelete(destination, destinationTag)
+  }
+
+  /**
+   *
```
Comment (keefertaylor) [inline src/XRP/model/xrp-account-delete.ts:31]: nit: bonus empty line

File: src/XRP/model/xrp-account-delete.ts [hunk 4]
```diff
@@ -0,0 +1,41 @@
+import { AccountDelete } from '../Generated/web/org/xrpl/rpc/v1/transaction_pb'
+
+/*
+ * Represents an AccountDelete transaction on the XRP Ledger.
+ * An AccountDelete transaction deletes an account and any objects it owns in the XRP Ledger,
+ * if possible, sending the account's remaining XRP to a specified destination account.
+ * @see: https://xrpl.org/accountdelete.html
+ */
+export default class XRPAccountDelete {
```
Comment (keefertaylor) [inline src/XRP/model/xrp-account-delete.ts:9]: I think that we should have a TODO to make this object use X-Address format too? WDYT?

File: src/XRP/model/xrp-account-delete.ts (hunk 1 above)
Comment (keefertaylor) [reply src/XRP/model/xrp-account-delete.ts:23]: The general view I've taken is we have to believe the server isn't malicious. 

In particular, the [documentation](https://xrpl.org/accountdelete.html) notes:
- destination is non-optional
- destinationTag is optional

File: test/XRP/fakes/fake-xrp-transaction-type-protobufs.ts [hunk 5]
```diff
@@ -25,6 +31,10 @@ const testSetFlag = 4
 const testTransferRate = 1234567890
 const testTickSize = 7
 
+// AccountDelete values
+const testDestination = 'rPEPPER7kfTD9w2To4CQk6UCfuHM9c6GDY'
```
Comment (tedkalaw) [inline test/XRP/fakes/fake-xrp-transaction-type-protobufs.ts:35]: super nit: seems like it's common in this file to put a space between the comment describing the values and the values themselves?

File: test/XRP/fakes/fake-xrp-transaction-type-protobufs.ts [hunk 6]
```diff
@@ -62,4 +72,26 @@ testAccountSetProtoAllFields.setTickSize(testTickSizeProto)
 const testAccountSetProtoOneFieldSet = new AccountSet()
 testAccountSetProtoOneFieldSet.setClearFlag(testClearFlagProto)
 
-export { testAccountSetProtoAllFields, testAccountSetProtoOneFieldSet }
+// AccountDelete protos
+const testAccountAddressProto = new AccountAddress()
```
Comment (tedkalaw) [inline test/XRP/fakes/fake-xrp-transaction-type-protobufs.ts:76]: same here - not sure which it's supposed to be

Comment (amiecorso) [reply test/XRP/fakes/fake-xrp-transaction-type-protobufs.ts:76]: good nit - this actually gets taken care of in a downstream PR I believe so I can be negligent here

**Pull request merged**